### ISSUE TITLE
Exporting a beatmap to legacy creates an overabundance of inherited timing points

### ISSUE THREAD

**Issue by irisu01** [ppy/osu#22171]
### Type

Game behaviour

### Bug description

Exporting a map from osu!lazer to import into stable creates an overabundance of timing points atop uninherited points (red points) as well as at the start and end of sliders.

### Screenshots or videos

![osu!_ICOUZsdfCS](https://user-images.githubusercontent.com/30084105/212238426-b8999274-7fd7-412a-88c4-913888bf9c74.png)
![osu!_rb2f7J8XUM](https://user-images.githubusercontent.com/30084105/212238540-3f0408ca-457c-4a60-99cc-3d1a4253b145.png)

### Version

2022.1288.0

### Logs

[legacy-ipc.log](https://github.com/ppy/osu/files/10408299/legacy-ipc.log)
[network.log](https://github.com/ppy/osu/files/10408300/network.log)
[performance.log](https://github.com/ppy/osu/files/10408301/performance.log)
[performance-audio.log](https://github.com/ppy/osu/files/10408302/performance-audio.log)
[performance-draw.log](https://github.com/ppy/osu/files/10408303/performance-draw.log)
[performance-input.log](https://github.com/ppy/osu/files/10408304/performance-input.log)
[performance-update.log](https://github.com/ppy/osu/files/10408305/performance-update.log)
[runtime.log](https://github.com/ppy/osu/files/10408307/runtime.log)
[updater.log](https://github.com/ppy/osu/files/10408308/updater.log)
[database.log](https://github.com/ppy/osu/files/10408309/database.log)
[input.log](https://github.com/ppy/osu/files/10408310/input.log)

**Comment by peppy** [comment]
The reasoning here looks to be the flipflop between "normal" and "drum" sample sets. Could you please include or link the beatmap used to test this?

**Comment by irisu01** [comment]
After further testing, it appears as if that was the issue. I imported a personal map from a friend, and was accidentally using hitnormals over drums. However, using `CTRL + A` to reassign every note to `drum` creates the same similar effect.

I've included a map directly exported from lazer where this occurs.

(Mai Zang - Si Ye Cao De Huang Xiang Irisu)[https://cdn.discordapp.com/attachments/234057206784458754/1063343143744974880/Mai_Zang_-_Si_Ye_Cao_De_Huan_Xiang_Irisu.osz]
